server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -476,6 +476,25 @@ public VolumeVO doInTransaction(TransactionStatus status) {
         });
     }
 
+    /**
+     * Retrieves the volume name from CreateVolumeCmd object.
+     *
+     * If the retrieved volume name is null, empty or blank, then A random name
+     * will be generated using getRandomVolumeName method.
+     *
+     * @param cmd
+     * @return Either the retrieved name or a random name.
+     */
+    public String getVolumeNameFromCommand(CreateVolumeCmd cmd) {
+        String userSpecifiedName = cmd.getVolumeName();
+
+        if (org.apache.commons.lang.StringUtils.isBlank(userSpecifiedName)) {
```

Comment:
@mike-tutkowski : Exactly there is one more StringUtils. If you put this it will conflict. I tried in that way but it gives error.